code/modules/paperwork/folders.dm

```diff
@@ -6,119 +6,114 @@
 	w_class = WEIGHT_CLASS_SMALL
 	pressure_resistance = 2
 	resistance_flags = FLAMMABLE
+	/// The background color for tgui in hex (with a `#`)
+	var/bg_color = "#7f7f7f"
 
 /obj/item/folder/suicide_act(mob/living/user)
 	user.visible_message("<span class='suicide'>[user] begins filing an imaginary death warrant! It looks like [user.p_theyre()] trying to commit suicide!</span>")
 	return OXYLOSS
 
-/obj/item/folder/blue
-	desc = "A blue folder."
-	icon_state = "folder_blue"
+/obj/item/folder/Initialize()
+	update_icon()
+	. = ..()
 
-/obj/item/folder/red
-	desc = "A red folder."
-	icon_state = "folder_red"
+/obj/item/folder/Destroy()
+	for(var/obj/important_thing in contents)
+		if(!(important_thing.resistance_flags & INDESTRUCTIBLE))
+			continue
+		important_thing.forceMove(drop_location()) //don't destroy round critical content such as objective documents.
+	return ..()
 
-/obj/item/folder/yellow
-	desc = "A yellow folder."
-	icon_state = "folder_yellow"
+/obj/item/folder/examine()
+	. = ..()
+	if(contents)
+		. += "<span class='notice'>Alt-click to remove [contents[1]].</span>"
 
-/obj/item/folder/white
-	desc = "A white folder."
-	icon_state = "folder_white"
+/obj/item/folder/proc/rename(mob/user)
+	if(!user.is_literate())
+		to_chat(user, "<span class='notice'>You scribble illegibly on the cover of [src]!</span>")
+		return
 
+	var/inputvalue = stripped_input(user, "What would you like to label the folder?", "Folder Labelling", "", MAX_NAME_LEN)
```

Comment:
```suggestion
	var/inputvalue = stripped_input(user, "What would you like to label the folder?", "Folder Labelling", null, MAX_NAME_LEN)
```

Reply:
I assume behind the scenes they both are null, but I cant help but get feeling something wont render properly

like the case wether its empty string or actuely nothing(null)